src/main/java/com/contentful/java/cda/CDAHttpException.java

```diff
@@ -37,6 +40,9 @@ public CDAHttpException(Request request, Response response) {
 
   private String readResponseBody(Response response) {
     try {
+      BufferedSource bufferedSource = response.body().source();
+      Timeout timeout = bufferedSource.timeout();
+      timeout.deadline(1, TimeUnit.SECONDS);
```

Comment:
Also not sure about the 1 second here, if you don't have a great connection, some requests may take a while. Maybe inject the timeout as a parameter? And have it default to 1 if not sent?